src/udev/udevadm-verify.c

```diff
@@ -124,7 +153,7 @@ int verify_main(int argc, char *argv[], void *userdata) {
                 const char* const* rules_dirs = STRV_MAKE_CONST(CONF_PATHS("udev/rules.d"));
                 _cleanup_strv_free_ char **files = NULL;
 
-                r = conf_files_list_strv(&files, ".rules", NULL, 0, rules_dirs);
+                r = conf_files_list_strv(&files, ".rules", arg_root, 0, rules_dirs);
                 if (r < 0)
                         return log_error_errno(r, "Failed to enumerate rules files: %m");
```

Comment:
Instead of the above check, please warn and exit here `if (arg_root && strv_isempty(files))`.

Reply:
okay, but if the reason why `files` is empty wouldn't be available, what would you suggest writing in the diagnostic message?

Reply:
How about to simply log that we find no .rules file under the specified root directory?
```c
if (strv_isempty(files))
        return log_error_errno(SYNTHETIC_ERRNO(ENOENT), "No udev rules file found.");
```

If you'd like to diagnose more, then how about the following?
```c
if (strv_isempty(files)) {
        if (arg_root && access(arg_root, F_OK) < 0)
                return log_error_errno(errno, "Failed to access \"%s\": %m", arg_root);

        return log_error_errno(SYNTHETIC_ERRNO(ENOENT), "No udev rules file found.");
}
```

Reply:
This is the message I've ended up with, hope it's good enough:
```
return log_error_errno(SYNTHETIC_ERRNO(ENOENT), "No rules files found in %s", arg_root);
```